src/_data/ghost.js

```
@@ -30,13 +36,47 @@ module.exports = async () => {
     )
   )
     .then(arr => {
-      console.log('Finished processing all posts');
+      console.log('Finished processing all Ghost posts');
       return arr.flat();
     })
     .catch(err => console.error(err));
+  const hashnodePosts = await Promise.all(
+    chunk(allHashnodePosts, batchSize).map((batch, i, arr) =>
+      piscinaHashnode.run({
+        batch,
+        currBatchNo: Number(i) + 1,
+        totalBatches: arr.length
+      })
+    )
+  )
+    .then(arr => {
+      console.log('Finished processing all Hashnode posts');
+      return arr.flat();
+    })
+    .catch(err => console.error(err));
+
+  const ghostPostSlugsSet = new Set(ghostPosts.map(post => post.slug));
+  const hashnodePostSlugsSet = new Set(hashnodePosts.map(post => post.slug));
+  const commonPostSlugs = [...ghostPostSlugsSet].filter(slug =>
+    hashnodePostSlugsSet.has(slug)
+  );
+
+  const posts = [...ghostPosts, ...hashnodePosts]
+    .filter(post => {
+      if (commonPostSlugs.includes(post.slug)) {
+        errorLogger({ type: 'duplicate', name: post.title });
+        return false;
+      }
+      return true;
+    })
+    .sort((a, b) => {
+      return new Date(b.published_at) - new Date(a.published_at);
+    });
```

This appears to be removing both posts when there was a duplicate.

But I think it's better to keep the older post (compared by published_at) and remove only the new post, so that existing articles don't get removed unintentionally.

---

This was the behaviour we wanted from what I could gather from the dev meeting 2 weeks back ie. not build articles if slug matches from both source and report it.

cc: @raisedadead

---

Yeah - it will be a hard problem but during the move the authors (or at least the publication leads will need to be on the watch and "fix" slugs). We can do a automated "notification" when this happens.

---

@raisedadead I'm still not sure why we want to drop both posts. I think the older post has already been published a long time ago in most cases, and may have been linked from somewhere, so we'd like to keep it.

---

To actively block a build and notify the editorial team to fix it ASAP. I want to reiterate that the build will fail even in our current Ghost setup. This has always been the case.

Slug clashes are presumably rare instances because the headlines are literally handed off by the editorial team. If a clash happens, it is a deeper problem that needs fixing.

Lastly, I want to emphasize that this check is only to be used during the transition phase, not before, not after.

---

@raisedadead Thank you for clarifying. I wasn't aware that the build will fail.

Excuse me if I'm missing something because of my lack of knowledge, but if we want to make the build fail when the slug clash happens, should we throw an error here?

Currently, it logs the title of the duplicate articles to the `duplicate-errors.log` but doesn't seem to throw an error.

---

Yes.

---

Thank you so much for clearing up my questions 🙏 

**Meeting notes**
When slug clashes happen, we want to:
- stop the build (in production)
- log which articles caused the problem